Fix flaky test_verify_id_token_bad_tokens failure

This recently caused a "broken" [build][1] (though I restarted it)

```python
======================================================================
FAIL: test_verify_id_token_bad_tokens (tests.test_jwt.PEMCryptTestsOpenSSL)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/google/oauth2client/tests/test_jwt.py", line 181, in test_verify_id_token_bad_tokens
    self._check_jwt_failure(jwt, 'Token used too early')
  File "/home/travis/build/google/oauth2client/tests/test_jwt.py", line 94, in _check_jwt_failure
    self.assertTrue(expected_error in str(exc_manager.exception))
nose.proxy.AssertionError: False is not true
-------------------- >> begin captured logging << --------------------
oauth2client.crypt: DEBUG: ... REDACTED ...
oauth2client.crypt: DEBUG: ... REDACTED ...
oauth2client.crypt: DEBUG: ... REDACTED ...
--------------------- >> end captured logging << ---------------------
```

[1]: https://travis-ci.org/google/oauth2client/builds/108176722

--- a/oauth2client/crypt.py
+++ b/oauth2client/crypt.py
@@ -89,12 +89,12 @@ def make_signed_jwt(signer, payload, key_id=None):
         header['kid'] = key_id
 
     segments = [
-      _urlsafe_b64encode(_json_encode(header)),
-      _urlsafe_b64encode(_json_encode(payload)),
+        _urlsafe_b64encode(_json_encode(header)),
+        _urlsafe_b64encode(_json_encode(payload)),
     ]
     signing_input = b'.'.join(segments)
 
-    signature = signer.sign(signing_input).rstrip(b'=')
+    signature = signer.sign(signing_input)
     segments.append(_urlsafe_b64encode(signature))
 
     logger.debug(str(segments))

